### PR TITLE
x-pack/filebeat/module/cisco: fix event.{outcome,type} handling

### LINKED ISSUE #29581: Filebeat module cisco asa ECS event.outcome / event.type wrong
Hi,

I'm using the cisco module to parse logs from an ASA. The ES ingest pipeline then sets also the fields for event.outcome which is fine. But it seems that the pipeline sets the outcome field different than what the documentation says.

https://www.elastic.co/guide/en/ecs/1.12/ecs-using-the-categorization-fields.html

Here is an example showing a firewall blocking a connection and the field event.outcome is set to `success`.
The ES pipeline sets the event.outcome field to `failure` in such a case:
Also the field event.type is set to `info` and should be set `connection`.

This look wrong to me. Any thoughts on this?

Example output from the pipeline (some fields removed / redacted):
```
{
    "_source": {
        "@timestamp": "2021-09-01T23:58:58.000Z",
        "@version": "1",
        "destination": {
            "address": "1.1.1.1",
            "ip": "1.1.1.1",
            "port": 23
        },
        "event": {
            "action": "firewall-rule",
            "category": [
                "network"
            ],
            "code": 710003,
            "created": "2021-12-21T21:10:41.874Z",
            "ingested": "2021-12-21T21:10:44.496078Z",
            "kind": "event",
            "outcome": "failure",
            "severity": 3,
            "type": [
                "info",
                "denied"
            ]
        },
        "log": {
            "level": "err"
        },
        "message": "%ASA-3-710003: TCP access denied by ACL from 2.2.2.2/35025 to RULENAME:1.1.1.1/23",
        "network": {
            "iana_number": 6,
            "transport": "tcp"
        },
        "observer": {
            "egress": {
                "interface": {
                    "name": "RULENAME"
                }
            },
            "type": "firewall",
            "vendor": "Cisco"
        },
        "related": {
            "ip": [
                "2.2.2.2",
                "1.1.1.1"
            ]
        },
        "source": {
            "address": "2.2.2.2",
            "ip": "2.2.2.2",
            "port": 35025
        },
        ....
    }
}
```

- Version: Elastic Stack 7.16.1 / Filebeat 7.15.1
- Operating System: Ubuntu 18.04
- Steps to Reproduce: Can also be seen in the tests: https://github.com/elastic/beats/blob/master/x-pack/filebeat/module/cisco/asa/test/asa.log-expected.json#L4987

### PR DESCRIPTION
**Pull request by efd6**
## What does this PR do?

This fixes handling of event.outcome and event.type field in Cisco ASA and FTD modules to more correctly match the field descriptions in ECS.

## Why is it important?

Currently the fields do not conform to ECS. This is #29581.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
~~- [ ] I have made corresponding changes to the documentation~~
~~- [ ] I have made corresponding change to the default configuration files~~
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ] Check that new golden values are sane.

## How to test this PR locally

Run `TESTING_FILEBEAT_MODULES=cisco MODULES_PATH=module mage -v pythonIntegTest` in x-pack/filebeat.

First commit is boring, being just the update for golden files to prune `@timestamp` fields.

## Related issues

- Closes #29581.

## Use cases

N/A

## Screenshots

N/A

## Logs

N/A

### CODE REVIEW

File: x-pack/filebeat/module/cisco/shared/ingest/asa-ftd-pipeline.yml [hunk 1]
```diff
@@ -383,7 +383,7 @@ processors:
       if: "ctx._temp_.cisco.message_id == '304001'"
       field: "event.outcome"
       description: "304001"
-      value: success
+      value: "allowed"
```
Comment (adriansr) [inline x-pack/filebeat/module/cisco/shared/ingest/asa-ftd-pipeline.yml:386]: I don't see _allowed_ or _dropped_ specified as valid values for event.outcome, only _success_ and _failure._ My understanding is that this should go into event.type. Am I missing something?

**Review by adriansr — CHANGES_REQUESTED** [review]

File: x-pack/filebeat/module/cisco/shared/ingest/asa-ftd-pipeline.yml (hunk 1 above)
Comment (efd6) [reply x-pack/filebeat/module/cisco/shared/ingest/asa-ftd-pipeline.yml:386]: These are temporary values that are used to communicate to the painless later what the combination of `outcome` and type `should` be. If state is squashed to a binary of success/failure it is not possible to reconstruct the possibility of a successful denial. I could have done if with a second variable, but this feels better.

Comment (efd6) [reply x-pack/filebeat/module/cisco/shared/ingest/asa-ftd-pipeline.yml:386]: These get sorted out here https://github.com/elastic/beats/pull/29698/commits/76fe29f3f6e543c25b81f7308690c772697691b7#diff-35840a6c4edae3a4fc97e0a1c84d5576fcfd9bb64faf12e7a6800347a5728cb1R1857-R1872. If you'd like I can add commentary explaining this.

**Review by adriansr — APPROVED** [review]
LGTM

**Pull request merged**